src/t_string.c

```diff
@@ -109,6 +109,28 @@ void setGenericCommand(client *c,
         return;
     }
 
+    /* If the milliseconds have expired, then we don't need to set it into the
+     * database, and then wait for the active expire to delete it, it is wasteful.
+     * If the key already exists, delete it. */
+    if (expire && checkAlreadyExpired(milliseconds)) {
+        if (found) {
+            int deleted = dbGenericDelete(c->db, key, server.lazyfree_lazy_expire, DB_FLAG_KEY_EXPIRED);
+            serverAssertWithInfo(c, key, deleted);
+            server.dirty++;
+
+            /* Replicate/AOF this as an explicit DEL or UNLINK. */
+            robj *aux = server.lazyfree_lazy_expire ? shared.unlink : shared.del;
+            rewriteClientCommandVector(c, 2, aux, key);
+            signalModifiedKey(c, c->db, key);
+            notifyKeyspaceEvent(NOTIFY_GENERIC, "del", key, c->db->id);
+        }
```

Comment:
This code flow is duplicated from expire.c could we abstract this out to server.c maybe and invoke the same from expire.c / t_string.c ?

Reply:
i do add a new function to reduce the duplicated in the very first version: https://github.com/valkey-io/valkey/commit/2b0295623. But i struggled with the name and don't like the diff very much, so came up with this.

Do you think the diff is better? i am seeking more votes.

Reply:
I like the abstraction, and would would name it something like `deleteExpiredKeyFromOverwriteAndPropagate`.